File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/model/message/outgoing/ErrorMessage.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.model.message.outgoing

import ch.nevis.mobile.authentication.sdk.react.model.error.ChannelError
import kotlinx.serialization.Serializable

@Serializable
data class ErrorMessage(
    override val operationId: String,
    val exception: ChannelError? = null
) : ChannelOutMessage()

File: ios/Model/Message/Outgoing/ErrorMessage.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

class ErrorMessage: ChannelOutMessage {
	// MARK: Properties

	var exception: ChannelError?

	// MARK: Initialization

	init(operationId: String, error: Error? = nil) {
		if let error {
			self.exception = error.asChannelError()
		}
		super.init(operationId: operationId)
	}

	// MARK: Encodable

	enum CodingKeys: String, CodingKey {
		case exception
	}

	override func encode(to encoder: Encoder) throws {
		var container = encoder.container(keyedBy: CodingKeys.self)
		try container.encodeIfPresent(exception, forKey: .exception)
		try super.encode(to: encoder)
	}
}
